commit 336d258e9fe9b66e8952b88638bfef9c04fe0fcc
Author: John Levon <john.levon@joyent.com>
Date:   2019-07-29T12:20:03+00:00 (2 months ago)
    
    OS-7906 buffer overflow in zoneadmd/zfd.c
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>

diff --git a/usr/src/cmd/zoneadmd/zfd.c b/usr/src/cmd/zoneadmd/zfd.c
index 8f4307a653..307d916154 100644
--- a/usr/src/cmd/zoneadmd/zfd.c
+++ b/usr/src/cmd/zoneadmd/zfd.c
@@ -561,7 +561,7 @@ ctlcmd_process(int sockfd, int stdoutfd, unsigned int *flags)
 	if (i == 0) {
 		goto fail;
 	}
-	buf[i+1] = '\0';
+	buf[i] = '\0';
 
 	if (strncmp(buf, "TIOCSWINSZ ", 11) == 0) {
 		char *next = buf + 11;
